torch/csrc/jit/python/python_sugared_value.cpp

```diff
@@ -217,6 +217,54 @@ std::shared_ptr<SugaredValue> PythonModuleValue::attr(
   return toSugaredValue(member, m, loc, /*is_constant=*/true);
 }
 
+ModuleValue::ModuleValue(
+    Value* self,
+    std::shared_ptr<ConcreteModuleType> concreteType)
+    : self_(self), concreteType_(std::move(concreteType)) {
+  auto containedTypeHint = concreteType_->getContainedTypeHint();
+
+  if (containedTypeHint) {
+    // For now, only dict type containedTypeHints are supported. Generate and
```

Comment:
I think it might be better to defer generating the Dict until the value is actually indexed into. Right now, we're eagerly creating the dict so that each time the module value is referenced we create a new dict. It also locates the code for type-checking indexing more where you'd expect to be, instead of the generic ModuleValue creation